engines/director/lingo/lingo-builtins.cpp

```diff
@@ -672,6 +672,37 @@ void LB::b_deleteAt(int nargs) {
 	}
 }
 
+void LB::b_deleteOne(int nargs) {
+	Datum indexD = g_lingo->pop();
+	Datum list = g_lingo->pop();
+	TYPECHECK3(indexD, INT, FLOAT, SYMBOL);
+	TYPECHECK2(list, ARRAY, PARRAY);
+
+	switch (list.type) {
+	case ARRAY: {
+		g_lingo->push(list);
+		g_lingo->push(indexD);
+		b_getPos(nargs);
+		int index = g_lingo->pop().asInt();
+		if (index > 0) {
+			list.u.farr->arr.remove_at(index - 1);
+		}
+		break;
+	}
+	case PARRAY: {
+		Datum d;
+		int index = LC::compareArrays(LC::eqData, list, indexD, true, true).u.i;
```

Comment:
I think this is not correct either

Reply:
The compareArrays call is the same as the one in b_getOne (matching on value), except the output index is then piped to delete_at. Misleading variable has been renamed.